### PR TITLE
Add log for fetched events

### PR DESCRIPTION
**Pull request by sunce86**
# Description

Related to oncall issue where our system failed to index some settlements/trades. The only reason I could think of at the moment is that it was some intermittent issue with the node which answered with empty Vec of events. All other error execution paths are already covered in the code.

### CODE REVIEW

File: crates/shared/src/event_handling.rs [hunk 1]
```diff
@@ -387,6 +387,11 @@ where
             {
                 match result {
                     Ok(e) => {
+                        tracing::debug!(
+                            "events fetched for block: {:?}, events: {}",
+                            blocks[i],
+                            e.len(),
```
Comment (squadgazzz) [inline crates/shared/src/event_handling.rs:393]: Do we need this log for non-empty events?

Comment (sunce86) [reply crates/shared/src/event_handling.rs:393]: Probably not. But I guess with this one we can also check if node returns also partial list of events (which should be super weird but hey)

**Review by squadgazzz — APPROVED** [review]

File: crates/shared/src/event_handling.rs (hunk 1 above)
Comment (fleupold) [reply crates/shared/src/event_handling.rs:393]: To save a bit on logging space I think we should not log empty blocks (by virtue of not having this log we know there were none)

**Review by fleupold — APPROVED** [review]
I'd also add some logs in `append_events` and `replace_events` inside the `EventStoring` implementation.

**Comment by sunce86** [comment]
> I'd also add some logs in `append_events` and `replace_events` inside the `EventStoring` implementation.

Not sure which ones would be useful for this specific error we had. Let's add those separately if needed.

**Pull request merged**